### DISCUSSION TITLE
Error when using Set-PnPUserProfileProperty

### DISCUSSION THREAD

**Discussion by grey0ut** [pnp/powershell#801] [Q&A]
I'm new to this module, but not Powershell. I'm exploring whether I can use the PnP module to automate syncing some AD attributes to Sharepoint Online User Profile App (UPA). 
All my searching seems to have lead me to the Set-PnPUserProfileProperty cmdlet. I was able to get connected and register the PNP management shell, but when I attempt to modify any user's profile properties I'm always greeted with this error and I can't find any reference to it online.

Any help is appreciated.

```
Set-PnPUserProfileProperty : This operation requires you to be managing your own data or have administrator privileges.
At line:1 char:1
+ Set-PnPUserProfileProperty -Account User@contoso.com -PropertyNam ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : WriteError: (:) [Set-PnPUserProfileProperty], ServerUnauthorizedAccessException
    + FullyQualifiedErrorId : EXCEPTION,PnP.PowerShell.Commands.UserProfiles.SetUserProfileProperty
```

**Comment by AndersRask** [comment]
The functionality works fine, but I have also experienced problems with Set-PnPUserProfileProperty and permissions, depending on how I set it. For example, if using certificates for a service principal, I got the above error, but if I switched to ClientSecret and App-Only SharePoint permissions, it worked fine (Tenant: FullControl, Social: FullControl in AppPermissionRequest)

What permissions are you using for this? If you plan to automate, service principal is the way to go compared with an account.

**Reply by grey0ut** [reply]
This is probably ignorance on my part. I'm just using credential authentication with a global admin account to test. I'm seeing this article:
https://docs.microsoft.com/en-us/sharepoint/dev/solution-guidance/security-apponly-azureacs
And thinking that maybe I need to look at those steps.

**Reply by grey0ut** [reply]
My current app permission XML:
```
<AppPermissionRequests AllowAppOnlyPolicy="true">
  <AppPermissionRequest Scope="http://sharepoint/content/tenant" Right="FullControl" />
  <AppPermissionRequest Scope="http://sharepoint/social" Right="FullControl" />
</AppPermissionRequests>
```
Now receiving an error that the app does not have permission to access profile information. 

**EDIT**
I see it now. My second scope should have ended in "/social/tenant". Adding that fixed it and I was able to successfully set a user profile property.
